src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import duke.tasks.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Class that contains the list of tasks
+ *
+ */
+public class TaskList {
+
+    /** ArrayList of tasks */
+    private ArrayList<Task> taskList;
+
+    /** Counter to track tasklist size */
+    private int counter = 0;
+
+    /**
+     * Constructor to initialize tasklist
+     *
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    /**
+     * Method to add task to tasklist
+     *
+     * @param task The task to be added to tasklist
+     */
+    public void add(Task task) {
+        if (!taskList.contains(task)) {
+            this.taskList.add(task);
+            this.counter++;
+        }
+    }
+
+    /**
+     * Method to delete task from tasklist
+     *
+     * @param index The index of the task in the tasklist
+     * @return Deleted task
+     */
+    public Task delete(int index) {
+        Task deletedTask = this.taskList.get(index - 1);
+        this.taskList.remove(index - 1);
```

Comment:
Perhaps you could combine these two lines into one? Since ArrayList.remove() method returns the element at the index which you removed.
```suggestion
        Task deletedTask = this.taskList.remove(index - 1);
```